### PR TITLE
Make error handler public

### LINKED ISSUE #24672: Make ExceptionHandler public
```C#
public Func<ProcessErrorEventArgs, Task> ExceptionHandler { get; set; }
```
#
Interesting - this was internal in T1. I don't recall a specific ask to expose this, so I think we can change this back to internal.

### PR DESCRIPTION
**Pull request by JoshLove-msft**
Fixes https://github.com/Azure/azure-sdk-for-net/issues/24672

### CODE REVIEW

**Review by jsquire — APPROVED** [review]

**Comment by JoshLove-msft** [comment]
> Can we test this?

Added a test

File: sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/tests/ServiceBusEndToEndTests.cs [hunk 1]
```diff
@@ -1296,6 +1310,27 @@ public class TestCrossEntityTransactionBatch
...
+        {
+            public static async Task RunAsync(
+                [ServiceBusTrigger(FirstQueueNameKey)]
+                ServiceBusReceivedMessage message)
+            {
+                // Delay so that the lock will be lost when trying to complete automatically.
+                // We can't simply throw an exception here as it would be swallowed by TryExecuteAsync call in the listener.
+                // This means that the exception handler will not be used for errors originating from the function. This is the same
+                // behavior as in V4.
+                await Task.Delay(TimeSpan.FromSeconds(30));
```
[the remark this answers is not in the document]
Comment (JoshLove-msft) [reply sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/tests/ServiceBusEndToEndTests.cs:1323]: Unfortunately, there isn't another way to trigger the error handler without resorting to reflection hacks.

[the remark this answers is not in the document]
Comment (JoshLove-msft) [reply sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/tests/ServiceBusEndToEndTests.cs]: Disposed client as you suggested.

**Review by pakrym — APPROVED** [review]

**Pull request merged**